### PR TITLE
Bug 1243140. Move MS locales to DB

### PR DESCRIPTION
**Pull request by jotes**

### CODE REVIEW

File: pontoon/base/models.py [hunk 1]
```diff
@@ -472,6 +472,25 @@ def get_top_instances(self):
...
+        help_text="""
+        Microsoft Translator maintains its own list of <a href="https://msdn.microsoft.com/en-us/library/hh456380.aspx">supported locales</a>.
+        Choose a locale from that list that's is the closest match or leave it blank to disable support for this feature.
+        """
+    )
+    ms_terminology_code = models.CharField(
+        max_length=20,
+        blank=True,
+        help_text="""
+        Microsoft Terminology uses locale codes standarized by ISO-639. Add a code of compliant locale or leave blank to disable
```
Comment (mathjazz) [inline pontoon/base/models.py]: *Microsoft Terminology uses locale codes standarized by ISO-639.* -> that's also the case for Pontoon. The difference is that in Microsoft Terminology [language codes currently include both the language and the country/region](https://www.microsoft.com/Language/en-US/Microsoft-Terminology-API.aspx).

I suggest you change the help text to:
*Microsoft Terminology uses language codes that include both the language and the country/region. Chose a matching locale from <a href="https://www.microsoft.com/Language/en-US/Search.aspx">the list</a> or leave blank to disable support for Microsoft terminology.*

File: pontoon/base/models.py [hunk 2]
```diff
@@ -472,6 +472,25 @@ def get_top_instances(self):
...
 class Locale(AggregatedStats):
     code = models.CharField(max_length=20, unique=True)
+
+    # Codes related to Microsoft products.
+    ms_translator_code = models.CharField(
+        max_length=20,
+        blank=True,
+        help_text="""
+        Microsoft Translator maintains its own list of <a href="https://msdn.microsoft.com/en-us/library/hh456380.aspx">supported locales</a>.
+        Choose a locale from that list that's is the closest match or leave it blank to disable support for this feature.
```
Comment (mathjazz) [inline pontoon/base/models.py]: ...*to disable support for this feature* -> *to disable support for Microsoft Translator*

File: pontoon/base/static/js/main.js [hunk 3]
```diff
@@ -472,32 +463,21 @@ var Pontoon = (function (my) {
           url: '/microsoft-terminology/',
           data: {
             text: original,
-            // On first run, check if target locale supported
-            check: (self.locale.msTerminology === undefined) ? true : false,
-            // Use Microsoft Terminology locale, Pontoon's might not be supported
-            locale: (self.locale.msTerminology === undefined) ?
-                    self.locale.code : self.locale.msTerminology
+	    locale: self.locale.ms_terminology_code
```
Comment (mathjazz) [inline pontoon/base/static/js/main.js]: Nit: bad indent.

File: pontoon/base/static/js/main.js [hunk 4]
```diff
@@ -472,32 +463,21 @@ var Pontoon = (function (my) {
...
+	      append({
+		original: this.source,
+		quality: Math.round(this.quality) + '%',
+		url: 'https://www.microsoft.com/Language/en-US/Search.aspx?sString=' + this.source + '&langID=' + self.locale.msTerminology,
+		title: 'Visit Microsoft Terminology Service API.\n' +
+		       '© 2014 Microsoft Corporation. All rights reserved.',
+		source: 'Microsoft',
+		translation: this.target
+	      });
+	    });
```
Comment (mathjazz) [inline pontoon/base/static/js/main.js]: Nit: bad indent.

File: pontoon/base/migrations/0098_move_ms_locales_to_db.py [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    u'th-th': u'th',
+    u'tr-tr': u'tr',
+    u'uk-ua': u'uk',
+    u'ur-pk': u'ur',
+    u'vi-vn': u'vi',
+    u'wo-sn': u'wo',
+    u'yo-ng': u'yo',
+
+    # Batch of locales that require a review.
+    u'az-Latn-az': u'az',
```
Comment (mastizada) [inline pontoon/base/migrations/0098_move_ms_locales_to_db.py]: It was lower case in settings: `az-latn-az`. That code is bullshit but for some reason microsoft uses this version and it is unique for microsoft.

**Pull request merged**